%%%%%%%%%%%% packages %%%%%%%%%%%%%%%%%%%%%%%%%%
\documentclass[hidelinks]{article}
\usepackage[letterpaper,margin=1.0in]{geometry}
\usepackage[utf8]{inputenc}
\pagenumbering{arabic}
\usepackage{authblk}
\usepackage{graphicx}
\usepackage[singlelinecheck=false]{caption} % singlelinecheck makes single line caption left aligned instead of centered
\usepackage{subcaption}
\usepackage{amsmath}
\usepackage[round]{natbib}
\usepackage{fancyhdr}
\usepackage{longtable}
\usepackage{booktabs}

\pagestyle{fancy}
\fancyhead[R]{\textbf{On population genomic simulations of new model and non-model species: a guide}}
% for figures
\usepackage{graphicx}
\graphicspath{ {Figures/} }


% for highlighting text
\usepackage{xcolor}
\usepackage{soul}

% bibliography
\usepackage[round]{natbib}   % omit 'round' option if you prefer square brackets
\bibliographystyle{plainnat}

% hyperlinks
\usepackage{hyperref}



%commands to format figure and table references in the supplement
\newcommand{\beginsupplement}{%
        \fancyhead[L]{Supplemental Material}
        \setcounter{table}{0}
        \renewcommand{\thetable}{S\arabic{table}}%
        \setcounter{figure}{0}
        \renewcommand{\thefigure}{S\arabic{figure}}%
     }
\newcommand{\stopsupplement}{%
        \setcounter{table}{0}
        \renewcommand{\thetable}{\arabic{table}}%
        \setcounter{figure}{0}
        \renewcommand{\thefigure}{\arabic{figure}}%
     }


% Avoid pandoc bug when there are lists in the body.
\providecommand{\tightlist}{%
\setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

\title{Expanding the stdpopsim species catalogue, and lessons learned from realistic genome simulations}
\author[1]{M. Elise Lauterbur}
\author[2]{Maria Izabel A. Cavassim}
\author[3]{Ariella L. Gladstein}
\author[4]{Graham Gower}
\author[5]{Georgia Tsambos}
\author[6,7]{Jeff Adrion} % moved from above at request
\author[8]{Arjun Biddanda}
% \author[6]{Saurabh Belsare}
% \author[6]{Victoria Caudill}
\author[9]{Jean Cury}
\author[]{Ignacio Echevarria}
\author[10,11]{Ahmed Hasan}
% \author[]{Xin Huang}
\author[12]{Leonardo Nicola Martin Iasi} % lastname Iasi
\author[13]{Jana Obšteter}
\author[14]{Vitor Antonio Corrêa Pavinato} % lastname Pavinato
\author[15,16]{David Peede}
% \author[]{Ekaterina Noskova}
% \author[]{Alice Pearson}
\author[17]{Manolo Perez}
\author[6]{Chris Smith}
\author[18]{Jeff Spence}
% \author[2]{Anastasia Teterina}
\author[19]{Per Unneberg}
\author[20]{Juan Manuel Vasquez}
\author[21]{Ryan Waples}
\author[22]{Anthony Wilder Wohns}
\author[23]{Yan Wong}
\author[24]{Reed Cartwright}
\author[25]{Aaron Ragsdale}
\author[26]{Franz Baumdicker}
\author[27]{Gregor Gorjanc}
\author[28]{Ryan Gutenkunst}
\author[23]{Jerome Kelleher}
\author[6]{Andy Kern}
\author[6,29]{Peter Ralph}
\author[30]{Dan Schrider}
\author[31]{Ilan Gronau}


\affil[1]{\small{Department of Ecology and Evolutionary Biology, University of Arizona, Tucson AZ 85719}}
\affil[2]{\small{Department of Ecology and Evolutionary Biology University of California, Los Angeles}}
\affil[3]{\small{Embark Veterinary, Inc., Boston, MA 02111, USA}}
\affil[4]{\small{Section for Molecular Ecology and Evolution, Globe Institute, University of Copenhagen, Denmark}}
\affil[5]{\small{School of Mathematics and Statistics, University of Melbourne, Australia}}
\affil[6]{\small{Institute of Ecology and Evolution, University of Oregon, Eugene OR 97402}}
\affil[7]{\small{Ancestry}}
\affil[8]{\small{54Gene, Inc., Washington, DC 20005, USA}}
\affil[9]{\small{Université Paris-Saclay, CNRS, INRIA, Laboratoire Interdisciplinaire des Sciences du Numérique, UMR 9015 Orsay, France}}
\affil[10]{\small{Department of Cell and Systems Biology, University of Toronto, Toronto ON}}
\affil[11]{\small{Department of Biology, University of Toronto Mississauga, Mississauga ON}}
\affil[12]{\small{Department of Evloutionary Genetics, Max Planck Institute for Evolutionary Anthropology, Leipzig, Germany}}
\affil[13]{\small{Agricultural Institute of Slovenia, Department of Animal Science, Hacquetova ulica 17, Ljubljana, Slovenia}}
\affil[14]{\small{Entomology Dept., CFAES, The Ohio State University, Wooster, Ohio}}
\affil[15]{\small{Department of Ecology and Evolutionary Biology, Brown University, Providence, RI, USA}}
\affil[16]{\small{Center for Computational Molecular Biology, Brown University, Providence, RI, USA}}
\affil[17]{\small{Department of Genetics and Evolution, Federal University of Sao Carlos, Sao Carlos 13565905, Brazil}}
\affil[18]{\small{Department of Genetics, Stanford University School of Medicine, Stanford, CA, 94305}}
\affil[19]{\small{Department of Cell and Molecular Biology, National Bioinformatics Infrastructure Sweden, Science for Life Laboratory, Uppsala University,  Husargatan 3, SE-752 37 Uppsala, Sweden}}
\affil[20]{\small{Department of Integrative Biology, University of California, Berkeley, Berkeley, CA, USA}}
\affil[21]{\small{Department of Biostatistics, University of Washington}}
\affil[22]{\small{Broad Institute of MIT and Harvard, Cambridge, MA 02142, USA}}
\affil[23]{\small{Big Data Institute, Li Ka Shing Centre for Health Information and Discovery, University of Oxford, OX3 7LF, UK}}
\affil[24]{\small{Genomics, Evolution, and Bioinformatics, School of Life Sciences and The Biodesign Institute, Arizona State University, Tempe, Arizona}}
\affil[25]{\small{Integrative Biology, University of Wisconsin-Madison, Madison, Wisconsin}}
\affil[26]{\small{Cluster of Excellence - Controlling Microbes to Fight Infections, Eberhard Karls Universität Tübingen, Tübingen, Baden-Württemberg, Germany}}
\affil[27]{\small{The Roslin Institute and Royal (Dick) School of Veterinary Studies, University of Edinburgh, Edinburgh EH25 9RG, UK}}
\affil[28]{\small{Molcular and Cellular Biology, University of Arizona, Tucson, Arizona 85721}}
\affil[29]{\small{Department of Mathematics, University of Oregon, Eugene OR 97402}}
\affil[30]{\small{Department of Genetics, University of North Carolina at Chapel Hill, Chapel Hill, North Carolina 27599}}
\affil[31]{\small{Efi Arazi School of Computer Science, Reichman University, Herzliya, Israel}}

\date{\small{\today{}}}

\begin{document}

\maketitle


\section*{Abstract}

Simulation is a key tool in population genetics, useful for both
methods development and empirical research. An avalance of population
genomic data is in progress, as data is being generated faster than ever before with
efforts such the Earth Biogenome and its affiliated project networks.
This data, coupled with advances in estimating detailed population genetic models, has made acute the need
for more detailed simulations of a wide range of scenarios and model and non-model species.
Many empirical researchers employing population genetics wish to simulate
their study species, but do not know what information is required in order
to do so sufficiently realistically, what is sufficient realism to address
their question, where to find this information, or how to best share it for ease of reproducibility.
In this paper we discuss the elements of a population genomic
simulation model, including the required input data to make
the model a realistic characterisation of a particular species. We also discuss
common pitfalls and major considerations in choosing this input data.
Further, we describe how new species models can be integrated into the
catalog of stdpopsim, a recently-developed tool that makes it easy to simulate
complex population genetic models using up to date information.
So far stdpopsim has been limited to well characterised model species 
such as humans, chimpanzees, and \emph{Arabidopsis},
and we illustrate the process of adding a species to stdpopsim using
examples and lessons learned from a recent hackathon designed to expand the range of supported species.
Thus, this paper provides a means to expand the accessibility of population
genetic simulations to the broader empirical and development population
genetic community by serving as a tutorial in both how to
assemble the data that is required to simulate
a species, and how this information can be incorporated
into the stdpopsim catalog to make it available and accessible to everyone.


\hypertarget{introduction}{%
\section*{Introduction}\label{introduction}}

Dramatic reductions in sequencing costs are enabling the generation of
unprecedented amounts and diversity of genomic data for a huge variety of species
\citep{Ellegren2014}. Ongoing efforts to systematically sequence life on
Earth by efforts such as the Earth Biogenome \citep{Lewin2022} and its
affiliated project networks (for example, Vertebrate Genomes
\citep{Rhie2021}, 10,000 Plants \citep{Cheng2018} and others, see
https://www.earthbiogenome.org/affiliated-project-networks) are
providing the backbone to facilitate enormous increases in population-level genomic data for
new model and non-model species. Methods for inferring
demographic history and natural selection from such data are also flourishing
\citep{Beichman2018}. Past methods development has justifiably focused on the
human genome, or a few key species such as \emph{Drosophila}, 
as model systems. More recently attention is being paid to
generalize methods to include important population dynamics not present
in these models, such as inbreeding \citep{Blischak2020}, skewed offspring
distributions \citep{Montano2016}, selfing (eg. as implemented in Demes
\citep{Gower2022}), intense artificial selection \citep{MacLeod2013,
MacLeod2014}, and to realistically model individual non-model species of interest.

Simulations from population genomic models are an important component
for analysing this new data and testing these models; simulations are thus vital to both methods
development and empirical research. For example, they provide training
data for inference methods based on machine learning \citep{Schrider2018} or
Approximate Bayesian Computation \citep{Csillery2010}. They can also serve as
baselines for further analyses: for example, models incorporating
demographic history serve as null models in selection analyses
\citep{Hsieh2016a} or to seed downstream breeding program simulations
\citep{Gaynor2020}. More recently, population genomic simulations have begun
to be used to help guide conservation decisions for threatened species
\citep{Teixeira2021}.

In general, the usefulness of population genomic simulations increases
the more realistically they represent the species being simulated---that
is, as they incorporate more relevant elements of the species' biology. Important
elements include genome features such as mutation and recombination
rates that strongly affect genetic variation and haplotype structure
\citep{Nachman2002}, particularly when linked selection is important \citep{Cutter2013}. 
Furthermore, the demographic history of a
species, encompassing population sizes, divergences, and gene flow, can
dramatically affect patterns of genetic variation \citep{Teshima2006}. Thus
estimates of these and other evolutionary parameters (e.g. those governing
the process of natural selection) are fundamentally important to the
development of simulations of species of interest. This presents
challenges not only in the coding of the simulations themselves, but in
the choice of parameter estimates to be used to shape the simulation
model.

stdpopsim is a community resource recently developed to provide easy
access to detailed population genomic simulations \citep{Adrion2020}. This
resource lowers the technical barriers to performing these simulations
and reduces the possibility of erroneous implementation of simulations
for species with published models. But so far stdpopsim has been
primarily restricted to well-characterized model species. Feedback from
stdpopsim workshops has emphasized the community desire to simulate
non-model species of interest and ideally incorporate these into the stdpopsim catalog,
the need for a better understanding among the empirical population
genomic community of when it is practical to create a realistic
simulation of a species of interest, and which genomic elements are
necessary and how to choose relevant parameter estimates for them.

The choice of whether and how to develop population genomic
simulations for a species of interest is affected by the intended
analysis and the genomic resources and knowledge available for the
species. These choices have a major impact on the resulting patterns of
genomic variation generated by the simulation. The fundamental
importance of these components of realistic population genomic
simulations is not always well understood, and the necessary choices can
be challenging. While stdpopsim provides a framework for standardizing
simulations of some species, the broader population genetics community
can benefit from additional guidance in making and implementing these
choices to simulate a species of interest.

This paper is intended as a resource for both methods
developers and empirical researchers to develop simulations of their own
species of interest, with the potential to submit the simulation
framework for inclusion in the stdpopsim catalog for peer review
(DRS: is peer review the right phrase for this? Maybe ``independent verification?'') and
community use. In the \textbf{Tutorial}, we discuss the elements of a
population genomic simulation model that realistically characterizes a
species, including when a whole-genome simulation is more useful than
simulations based on either individual loci or generic loci, the required input data (genome
assembly, mutation and recombination rates, and demographic model) and
its quality, common pitfalls in choosing appropriate parameters, and
considerations for how to approach species that are missing some
necessary inputs. This paper is not intended as a tutorial for
implementing simulations in any particular simulator, rather to provide
guidance for what information is sufficient for a realistic simulation
using any simulator, and how to add this information to stdpopsim. The latter 
is discussed in the \textbf{Application} section, where we show how
species models may be integrated into the stdpopsim catalog. This includes
briefly presenting the current method for adding species, clarifying the
required genomic resources, and describing the quality control process
that reflects the peer review of a species model.
(DRS: I am not in love with the changes I have suggested here,
but I felt like in the current version of the text, the Application
part seemed like an afterthought when really it is the crux of the paper
and the titular section, given that this is the ``adding-species-manuscript''!)


\hypertarget{sec2}{%
	\section*{Standardized genome-wide simulations}\label{sec:std-sim}}

\noindent\emph{Ilan: In this section I condensed and reordered the text that was previously
	in ``The importance of independent quality control'' and ``The importance of simulating the whole genome". The second part is especially shortened, so people should examine this to make
	sure that nothing important got cut.  The Tex source contains a more detailed documentation
	of parts removed. The ending of the section requires clarification by Andy or Ryan.\vspace{5pt}}



%%%%%%%%%%%%%%%%
%ILAN: para 1 - give a broad intro to what is involved in popgen simulations.
%   We later dive into the details in section 4.
%%%%%%%%%%%%%%%%
% ILAN: the sentence below looks like it would fit well in the intro,
%   so I'm commenting it out here but will later consider moving it
%
%Simulation is one of the key tools in population genetics, but can
%present unexpected challenges and has many hidden pitfalls for the
%unwary population geneticist. 

Implementing a useful population genomic simulation requires setting numerous
parameters that specify genome architecture and evolutionary history.
Broadly speaking, these parameters can be divided into the following categories:
%
\begin{enumerate}
	\item The lengths of simulated segments
	\item Mutation rates
	\item Recombination rates
	\item Demographic history (ancestral population sizes, divergence times, and migration rates)
	\item Landscape of selection (e.g., sites of elements under selection, and a distribution of fitness effects)
\end{enumerate}
%
Generating realistic simulations first of all requires obtaining reliable estimates of
the different parameters, which is typically done by careful review of the relevant literature. The parameters then need to be applied correctly as input of a
simulation software, such as \texttt{msprime} \citep{Kelleher2016,Nelson2020} or \texttt{SLiM} \citep{Haller2019}. This step often involves non-trivial conversion
between different scales. Finally, the simulations are executed, which may
require considerable computation resources.


%%%%%%%%%%%%%%%%
%ILAN: para 2 - main advantages of standardized simulations
%  I took the relevant text from the "working together" section
%%%%%%%%%%%%%%%%
%(DRS: suggestion for a title that I think is more specific:
%``The importance of independent quality control''. But just a suggestion!)
%ILAN: this is no longer a section with a title. Not sure it needs one

The main objective of \texttt{stdpopsim} is to standardize this process, focusing on the
first two steps of curating the relevant literature and applying parameters
correctly to the simulation software. This standardization has several key advantages
\citep{Adrion2020}. First, it makes it easier to compare between different
simulations generated for the same species. Another major advantage of the
\texttt{stdpopsim} framework is the quality control process, by which the parameter choices and their conversion to the relevant units are checked by multiple researches.
This process involves independently creating a separate set of simulation
parameters based on the same literature sources. The new set of parameters are then
compared to the original ones by automated scripts. If discrepancies are found, they
are resolved between the two researchers with input of additional members of
the community. This process quite often
finds subtle bugs  \citep{Ragsdale2020} or highlights parts of the model that are
ambiguously defined by the literature sources. 
Therefore, this quality control considerably increases the reliability of the
resulting simulations in any downstream analysis.

% ILAN: I commented this out because it seems implied by the above
%   arguments. We might want to return to this in the discussion and
%   emphasized that some of the standards we use in stdpopsim can be used
%   by simulations, even if they are not done as part of our community.
%
%However the advantage working together to validate population genomic
%model implementation is not restricted to species that are added to
%the stdpopsim catalog. Researchers implementing population genomic
%simulations are encouraged to have their code and parameter choices
%checked by at least one other person before using it to create
%simulations. In addition, basic genomic features of the simulation
%results (e.g. the site frequency spectrum, the extent of linkage disequilibrium, etc.)
%can be compared to the reference genome and/or known population
%genetic characteristics. In this too it is useful to have the input
%of another researcher with fresh eyes.


%%%%%%%%%%%%%%%%
%ILAN: para 3 - advantages of whole-chrom simulations
%  I took the relevant text from the "simulating the whole genome" section
%  I shortened wuite a bit, trying to focus on the main points.
% In particular, I avoid mentoining some of the caveats:
% - no reference genome
% - no linkage across chroms
% We may wish to mention these in the discussion as future work, but I thought
% that mentioning them here was too distracting.
%%%%%%%%%%%%%%%%
One of the things we advocate in \texttt{stdpopsim} is simulation of complete
chromosomes. Such simulations are typically more complex because they require
setting many parameters pertaining to genome structure, and they require
considerable computational resources. Because of this, most simulations carried
out for individual studies avoid simulating complete chromosomes and resort to
separate simulations of many short segments. While this results in simpler
simulations that can be executed concurrently, % and thus much more efficiently, 
simulations of short segments ignore the structure of physical linkage along
chromosome. This linkage affects patterns of variation in subtle ways, which could
have significant influence on different aspects of downstream population genetic
analysis \citep{Nelson2020}. 


% ILAN: commented out part about modeling variation and not the exact sequence
%
%For instance, we should not ask that a simulation have
%polymorphic sites at precisely the same genomic positions observed in a
%real dataset, since it is these patterns of polymorphism themselves that
%we want to compare to the real data. Similarly, most simulations today
%do not use a reference genome sequence, since most population-level
%summaries of genetic variation do not take into account nucleotide
%identity. (However, this may change in the future.
% (DRS: I am not sure we need this text. Perhaps we could just start with
% the ``what are the benefits of whole-genome simulation'' question below?)
% ILAN: I agree. Commenting this out for now. May be something we can
%   return to in the discussion.

First, the correlation between different sites along a chromosome reduces
variation. This has a particularly striking effect in long stretches of
low recombination rates, as observed on human chromosome 22 \citep{Dawson2002}.
Second, the lengths of shared haplotypes within and
between populations provides valuable information about recent demographic history.
Methods that use such information, such as MSMC \citep{Schiffels2020}, require as
input very long genomic segments, with realistic recombination rates.
Thus, even when conducting simulations under neutral evolution, short segments
would not be able to capture important features observed in neutral regions of
real genome sequences.
%
%The effect of natural selection on patterns of inheritance would not have
%been evident if the chromosome had been broken into many independent pieces
%for simulation.
%(TODO: I think we lost something here - where/when would these effects not
%have been evident?)
%ILAN: I think this is fine now.
%
%
When conducting simulations with natural selection, linkage has
an even stronger effect. Selection acting on a small number of sites drives down
genetic variation in nearby sites. This has been argued to be one of the causes
for the observed correlation between recombination rate and genetic diversity
\citep{Begun1992}. Selection at linked sites has been shown to have a widespread
effect on patterns of genome variation in different species \citep{McVicker2009,Charlesworth2012}. This effect tends to be stronger in species
with smaller effective population sizes, but recent selective sweeps can dramatically
reduce variation even in species with large population sizes \citep{Lynd2010}.
%
% ILAN: commented out this part on genetic load. We can mention this briefly,
%   if we think it's important
%
%Third, linkage has the potential to affect demographic realism. The
%genetic load in a simulation of a small segment of chromosome with
%deleterious mutations will necessarily be less than that in whole
%chromosome. This situation makes it easy to simulate unrealistically
%high levels of load without realizing it, if only small unlinked segments are used.
%(TODO: What is the direct effect on demographic realism of the effect of
%load?) (DRS: I think that the three examples given in this section
%aren't the most compelling. The point of the chr22 example isn't super clear,
%the chromosome ends thing seems like a minor issue especially given that
%rec rates there are generally pretty low, and the load case seems to be
%more an issue of ``small scale sims let you simulate unrealistic secnarios''
%rather than ``small scale sims prevent you from simulating realistic scenarios''
%

% ILAN: not sure about this last part. Can someone summarize what
Finally, the whole genome sequence can provide important comparative
data with which to evaluate the simulations. This is a powerful tool for
making sure the simulations correspond to the elements of biological
realism that are important for the intended analyses \citep{Hsieh2016a}. 
(@Andy and/or @Ryan, or anyone else, can you expand on this?)
(ILAN: I didn't remove this part, because I know it's supposed to be a compelling example,
but I think that this requires more explanation).
%
%ILAN: end with statement that one still needs to think if this is 
%  required for certain analysis.
That said, for some purposes it might be completely fine to simulate short sequence
segments or even independent (unlinked) sites. However, this should be done after
careful consideration  of the benefits of whole-chromosome simulations.


% ILAN: commented out this final caveat. We can mention this briefly in the
%   end of this section, or just return to it in the discussion.
%
%Finally, what about \emph{whole genome} simulations? Chromosomes
%segregate independently, so between-chromosome correlations are generally close
%to zero. But they can be occur in fairly extreme situations, such as intense
%directional or stabilising selection on multiple loci across chromosomes
%\citep{Bulmer1971, Lara2022}. However, this situation can be simulated in
%follow-up forward-in-time simulations \citep{Haller2018, Gaynor2020}. For
%this reason, we tend to simulate chromosomes independently, and few
%simulators have mechanisms to simulate
%multiple chromosomes simultaneously. 
% DRS: SLiM kind of lets you do this...
% ILAN: I decided to remove this for now.

\hypertarget{sec3}{%
	\section*{The expanded \texttt{stdpopsim} catalog}\label{sec:expanded-catalog}}

\noindent\emph{Ilan: I moved relevant text about the updates we made to the catalog here.
	This requires another good pass and we can mention additional things like
	gene conversion.\vspace{5pt}}

stdpopsim is a community resource that makes detailed, realstic
population genomic simulations accessible \citep{Adrion2020}. When first
published, the stdpopsim catalog included 6 species: \emph{Homo
	sapiens}, \emph{Pongo abelii}, \emph{Canis familiaris}, \emph{Drosophila
	melanogaster}, \emph{Arabidopsis thaliana}, and \emph{Escherichia coli.}
We have since incorporated more demographic scenarios for \emph{Homo sapiens},
\emph{Pongo abelii}, \emph{Drosophila melanogaster}, and
\emph{Arabidopsis thaliana}. However, stdpopsim has the potential to be
useful to a wider community of researchers by involving the community
in adding new model and non-model species to the catalog.
To that end, the popsim consortium has made a concerted effort to 
involve the population genetics community in adding new species 
to the stdpopsim catalog.

In April 2021, we held a ``Growing the Zoo'' hackathon 
alongside the 2021 ProbGen conference. 
The hackathon was developed in response to participant demand from a
series of ``Introduction to stdpopsim'' workshops held in December 2020 and
February 2021. These workshops had over 150 participants, many of whom
expressed a wish to add their own species of interest to stdpopsim.
These workshops were followed by a ``Growing the Zoo'' workshop in
March 2021, to prepare interested participants for the hackathon. This
workshop introduced the process of developing a new model (including
parameter choice, as described above) and adding it to the stdpopsim
code base. All members of the population genetic community who were 
familiar with the this process, either from attending the
``Growing the Zoo'' workshop or their own previous work, were invited to
the hackathon. The hackathon ultimately added 12 species models to
the stdpopsim catalog, and three additional species have been added by
community members outside of the hackathon. (Figure \ref{fig:tree}),
and started the ball rolling on expanding the
stdpopsim catalog further. However not all species models that were started
at the hackathon were added, as we learned that there is a disconnect
between what species community members wish to simulate, and those
species that have sufficient resources for a realistic simulation.


%ILAN: took this from the "What about species lacking chromosome-level" section.
When we set out to cast a wide net and add a wide variety of species to
the catalog, we quickly ran into species that people were enthusiastic
to add, but lacked many (or most) of the parameters estimates discussed above. The
utility of stdpopsim is to make complex, multifaceted population genomic models easily available for
simulation; such data includes genetic maps, annotations, and/or
demographic models. We have not yet encountered a species with
widely-used demographic models but no chromosome-level assembly, so the
main issue in practice seems to be around chromosome-level assemblies
and around matching genome parameters to demographic models.

However, there is no clear line for what level of assembly quality is
required to be ``useful'' - the most telling indication is whether there is
a community of users eager to use it.

%ILAN: took this from the "Adding species to stdpopsim catalog" section.
The feedback from the workshops (2020-2021) made
it clear that many prospective users of stdpopsim want to simulate
non-model species that are not already in the stdpopsim catalog.
Currently there are large gaps in the species catalog (Figure \ref{fig:tree}), 
and having a greater representation of the tree of life would benefit 
the stdpopsim community as a whole.


\begin{figure}
	\includegraphics[width=\linewidth]{./figs/species_fig.png}
	\caption{Phylogenetic tree of species with models currently available in stdpopsim. 
		In blue are species in the original release, in orange are species added 
		by the community using the process described here. (DRS: maybe add another
		color to distinguish between those added during the hackathon and the other three.)}
	\label{fig:tree}
\end{figure}


\hypertarget{sec4}{%
	\section*{Guidelines for implementing a population genomic simulation}\label{sec:sim-guidelines}}

\noindent\emph{ILAN: I kept this part unchanged, including subsections.
	I plan to dive in during the next round of edits.\vspace{5pt}}


When choosing to generate appropriately realistic population genetic
simulations for a species of interest, certain population- and species-specific
parameters should be known. Regardless of the simulation program used,
for analyses that benefit from whole-genome simulation (see above), 
the current standard of realism ideally includes:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  a chromosome-level \textbf{genome assembly} (i.e. the lengths of each
  chromosome/scaffold researchers may wish to simulate),
\item
  an estimate of \textbf{mutation rate},
\item
  a \textbf{genetic map}, or at least an estimate of overall \textbf{recombination rate},
\item
  an estimate of \textbf{generation time}, (DRS: this should not always be required)
\item
  at least one \textbf{demographic model} (or at minimum the effective
  population size of a single-population),
\item
  a \textbf{genome annotation} showing at least coding and non-coding
  regions, and
\item
  a \textbf{distribution of fitness effects} for mutations occurring in
  various annotated regions.
\end{enumerate}

Any simulation requires \emph{some} choice for 1-5, although values may
have to come from another closely-related species (though this must be
an informed, not arbitrary, choice). A fully neutral simulation does not
require genome annotation nor a distribution of fitness effects, but
these are important for simulations of selective processes. Furthermore,
all of these parameters should be documented, citeable, and
ideally chosen after discussion with another researcher. (DRS: Why? 
What kind of researcher? And what should the nature of this discussion be? 
I am not sure I agree with this, really. The goal should be to accelerate
research by making it easier for researchers to simulate genomes, but if
they have to consult with colleagues about most of the choices they make
when designing the simulations maybe that somewhat defeats the purpose?)

Here we discuss all of these in more detail.

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  A \textbf{genome assembly}, with contigs assembled at the chromosome
  level or nearly so. By using chromosome level information to simulate
  genomes, we can indirectly simulate the effect of linkage and linked
  selection, as discussed above. Briefly, this includes more variation
  in simulation outcomes and reduced diversity around sites of selective
  sweeps, in part because the length of the resulting haplotypes and effects on
  chromosome ends can vary. The simulation of larger chromosome-length regions
  also allows for more accurate modeling of background selection, because
  diversity levels at a locus can be affected by even very distant purifying
  selection.

Although few species currently have truly chromosome-level assemblies,
we hope that many more will be available in the near future.
This is on the horizon thanks to the advent of long-read sequencing technologies
\citep{Amarasinghe2020}, and concomitant advances in assembly pipelines
\citep{Chakraborty2016}, as well as genome initiatives such as the Earth Biogenome
\citep{Lewin2022} and its affiliated project networks (for example,
Vertebrate Genomes \citep{Rhie2021}, 10,000 Plants \citep{Cheng2018} and others,
see https://www.earthbiogenome.org/affiliated-project-networks).

\item
  It is also required that the species has a plausible and citeable
  \textbf{mutation rate} estimate. Mutation rate estimates based on phylogenetic
  data, mutation accumulation studies, or sequence data from pedigrees are
  useful. Lacking one of these, it is common to use an estimate from
  another, closely related, species.
\item
  Since recombination also shapes the genetic diversity of genomes,
  simulations also require a \textbf{recombination rate} estimate.
  Ideally, this should be a chromosome-level \textbf{recombination map}, which
  allows more precise simulation of the effects of linked selection, the
  strength of which will vary dramatically across the genomes with highly
  heterogeneous recombination maps. At minimum, a citeable single recombination rate
  estimate for the whole genome is necessary. Lacking this from the species
  of interest, it is common to use a recombination rate estimate from
  another, closely related, species.
\item
  A \textbf{generation time} estimate is an important part of natural
  history that is required to translate results into real time units
  (eg. years), particularly for historical demographic models. This
  estimate is not generally an element of the simulations
  themselves because these typically work in units of generations, but
  used to adjust the simulation output.
\item
  The simulated species should have at least one citeable
  \textbf{demographic model}. This model should be appropriate for
  comparison with the intended population of study, as population
  histories can vary across the species' range. This specification of
  population history is a primary determinant of its levels of genetic
  diversity. Misspecification of a demographic model can generate highly
  unrealistic patterns of genetic variation \citep[e.g.,][]{Navascues2009}. At a
  minimum, simulations require a plausible and citeable estimate of 
  \textbf{effective population size}. A single-population simulation with a
  reasonable constant effective size should at least give comparable levels of
  overall genetic diversity. However, if there were sizable changes in
  effective population over time \citep{MacLeod2013}, these dynamics
  will have to be incorporated to generate meaningful patterns of
  genetic diversity. For example, a recent population expansion will produce
  an excess of low frequency alleles that no simulation of a constant-sized
  population will reproduce.
\end{enumerate}

The above factors are sufficient for modeling neutral variation in
populations, for example to infer past migration events or population
size changes under an assumption of neutrality. However, the goal of
much population genetics work is to understand the action or
consequences of selection. Although analytical models tend to study the
effect of single loci under selection in isolation, it has been
demonstrated that the effects of linked selection can vary substantially
along the genome in many species \citep{Wolf2017}, although even the general
strength and nature of this selection is still unknown. To generate
realistic amounts of selection at many locations on the genome, the
simulator needs to know where the selected sites occur, and what is the
nature of selection on them.

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{5}
\item
  The location of selected sites is conveyed by a \textbf{genome
  annotation}, such as a GFF3/GFF file containing
  information about the coordinates of coding and non-coding regions,
  and the position of specific genes.
\item
  A \textbf{distribution of fitness effects} (DFE) describes the relative
  frequencies of deleterious, neutral, and beneficial mutations. This
  distribution is important for understanding the rate of adaptive
  evolution. Not all simulators are able to incorporate this
  information; those that do include the general purpose forward-in-time
  population genetics simulator SLiM \citep{Haller2019}. The DFE,
  too, should be citeable and drawn from the species of interest or one
  that is closely related.
\end{enumerate}

\hypertarget{additional-considerations}{%
\subsubsection*{Additional
considerations}\label{additional-considerations}}

A commonly-encountered thorny issue is that estimates of some of these
parameters are interrelated with others. For instance, a published
demographic model may have been estimated assuming a generation time and
mutation rate that differ from today's understanding of the best guess
at those values. In such cases, what should be done? Naively using the
demographic model with a new estimate of mutation rate may lead to
unrealistic levels of genetic diversity. In some situations the
difference may come down to only a time scaling, so that in principle a
fitted demographic model may be translated to an equivalent one that
uses a different generation time or mutation rate. However, that process
would be opaque and fraught with potential for error.

When possible, all rate, generation time, and demographic models should
be drawn from the same study. Mixing sources is not ideal, but sometimes
necessary and should be approached with deliberation. In this situation
it may be helpful to discuss these choices and their potential
consequences with other researchers.
(TODO: we should explicitly discuss the task of lifting over from one build to
another. This seems like a good place for this.)

\hypertarget{what-if-we-dont-know-everything}{%
\paragraph{What if we don't know
everything?}\label{what-if-we-dont-know-everything}}

In our experience, a commonly encountered missing ingredient for a
population genetic simulations (neutral and non-neutral) in non-model
species is a chromosome-level assembly. Many species' genome assemblies
are composed of many relatively small contigs whose relation to each
other is unknown. There are two main ways to deal with this.

One option would be to effectively treat these contigs as chromosomes,
simulating them either together (unlinked) or separately. This feels
``realistic'', as the simulated chromosomes exactly match (in length, at
least) the real data. The alternative is to leave aside such strict
matching, instead simulating an anonymous chromosome from which patterns
of genetic variation can be extracted (if important, in chunks of size
similar to the contigs). The latter is usually more realistic,
since this includes linkage between sites that share a chromosome but
may be on different real contigs. As noted, this can affect patterns of neutral genetic
diversity and perhaps more crucially, linked selection. Contig-level
assemblies are sometimes annotated, and simulations of regions around
specific genes or genomic features (eg. exons) may be of interest.
However using the precise location of genes on many, unlinked contigs is
a false precision because of the importance of modeling linked selection
across contigs. For this reason, the question of ``realism'' is most
relevant for those species having chromosome-level assemblies. In our
experience it is uncommon that a species have widely-used demographic
models but not a chromosome-level assembly.

(DRS: I kind of got lost in the above paragraph, especially this part and the
few sentences after: ``instead simulating an anonymous chromosome from which patterns
of genetic variation can be extracted (if important, in chunks of size similar
to the contigs)''. In this part, are we talking about randomly assembling contigs
into chromosomes? Or adding sequence onto the ends of contigs to make each of them
into larger chormosomes? In the former we can have larger-scale linked selection
and all that good stuff, but if we extract data from each ``contig'' in our simulated
chromosome we have to be aware that the linkage these contigs experience with each
other differs from that of the real genome, since our assembly was arbitrary. We
don't have to worry about this with the latter strategy because we only sample one
``contig'' from a simulated chromosome, although we have to think about what to put
on the rest of that chromosome.)

Nonetheless, some of these ingredients will likely be missing for many
non-model species. Table 1 displays some considerations and solutions
For example, if all the parameters above are available except for a
chromosome level assembly, then one might choose to include only the
larger contigs.


% Please add the following required packages to your document preamble:
% \usepackage{booktabs}
\begin{table}[]
    \captionof{table}{Guide to missing parameters.} \label{tab:title} 
    \begin{tabular}{@{}llll@{}}
    \toprule
    Missing parameter                        & Options                                                                          & Considerations                                                                                                                                       &   \\ \midrule
    \multicolumn{1}{|l|}{Mutation rate}      & \multicolumn{1}{l|}{borrow from closest relative* with a citeable mutation rate} & \multicolumn{1}{l|}{will affect levels of polymorphism}                                                                                              &   \\ \cmidrule(r){1-3}
    \multicolumn{1}{|l|}{Recombination rate} & \multicolumn{1}{l|}{borrow from closest relative with a citeable rate}           & \multicolumn{1}{l|}{will affect patterns of selection, linkage, and linked selection}                                                                & | \\ \cmidrule(r){1-3}
    Demographic model                        & At least Ne is required and is estimable from mutation rate and genetic data     & the demographic history (e.g. bottlenecks, expansions, admixture) affects patterns of variation substantially {[}CITE{]}, a constant Ne is not ideal &   \\ \bottomrule
    \end{tabular}
\end{table}

\hypertarget{sec5}{%
	\section*{Examples of added species}\label{sec:std-sim}}

\noindent\emph{Ilan: I kept only the two examples here. The idea would be to
use these two species to describe different aspects of adding species: setting up parameters based on literature (cow) and the technical steps involved in adding species.  (mosquito).
I made very basic edit here and plan to get back to this later.\vspace{5pt}}

\hypertarget{bos-taurus}{%
	\subsubsection*{\texorpdfstring{\emph{Bos
				taurus} (cattle)}{Bos taurus (cattle)}}\label{bos-taurus}}

We use \emph{Bos taurus}, a species worked on during the hackathon, to
illustrate the process of choosing a species to simulate and finding and
deciding among the required resources.
%
%ILAN: this will be covered earlier
% We also use the lessons from the hackathon about which species are impractical to simulate
% to further discuss the limitations of population genomic simulations to
%realistically model species of interest when these species have
%inadequate genomic resources. However, we note that what it means for genomic resources to
%be inadequate must be approached in the context of the research goals,
%and in some situations simulation may be a useful tool for researchers
%studying organisms with fewer genomic resources.
%
\emph{Bos taurus} (cattle) was added to the stdpopsim catalog 
because of its agricultural importance. Agricultural species experience
strong selection due to domestication and selective breeding, leading
to a reduction in effective population size. These processes,
as well as admixture and introgression, produce patterns
of genetic variation that can be very different from typical model
species \citep{Larson2013}. In cattle, these processes have occurred over a
relatively short period (\textasciitilde 10,000 years or less) and are
increasingly intensified to improve food production \citep{Gaut2018,
MacLeod2013}. In cattle, high quality genome assemblies are now
available for several breeds \citep[e.g.,][]{Rosen2020, Heaton2021,
Talenti2022} and population genomic analyses have become widely used to
improve selective breeding and genomic prediction \citep{Meuwissen2001,
MacLeod2014, Obsteter2021}. Their small effective population size
(\textasciitilde 90 around 1980, and continuing to decline due to intense
selective breeding \citep{MacLeod2013, VanRaden2020, Makanjouloa2020}) is
challenging for demographic and selection inference \citep{MacLeod2013,
Hartfield2022} and genome-wide association and prediction
\citep{MacLeod2014}. For these reasons, it was useful to develop a
cattle model for stdpopsim.

With respect to the parameters chosen in the stdpopsim implementation,
for the basic genome simulation, we used the most recent assembly
\citep{Rosen2020}, a mutation rate of \(1.2 \times 10^{-8}\) \citep{Harland2017},
a recombination rate of \(0.926 \times 10^{-8}\) \citep{Ma2015}, a base population
effective population of 90 \citep{MacLeod2013}, and a generation interval of 5
years \citep{MacLeod2013}. The chosen effective population size is specific
to the Holstein breed, but is generally representative of a small
effective population size in other breeds. Critically, this low
effective population size will generate very low levels of overall
genetic diversity, which is not in line with the actually observed
variation \citep[e.g.,][]{Rosen2020}. To remedy this, the basic genome
simulation must be complemented with a demographic model. We implemented
the \cite{MacLeod2013} demographic model for the Holstein breed, which was
inferred from runs of homozygosity in the whole-genome sequence of two
iconic bulls. The estimated effective population size in this demography
spans from the deep past (33,154 generations ago with effective population size
of 62,000) to the recent past (about 1980 with effective population size of 90),
after which further breeding simulations with intense selection breeding are
needed to reflect reality \citep[e.g.,][]{
MacLeod2014, Gaynor2020, Obsteter2021}. \cite{MacLeod2013} assumed
recombination and mutation rates of \(10^{-8}\) in inferring their
demographic model, but revised the mutation rate to \(0.94 \times 10^{-8}\) by
taking sequence errors into account. In line with the advice given in
previous sections, we implemented these mutation and recombination rates
for the \cite{MacLeod2013} demographic model, even though we have implemented
more recent estimates in the basic genome model. 
When a simulation with this demographic model is requested, stdpopsim uses the
estimates assumed in the demographic model rather than the more
recent estimates, in line with our advice of using estimates from the same
source when possible.

\hypertarget{ano-gambea}{%
	\subsubsection*{\texorpdfstring{\emph{Anopheles gambiae} (mosquito)}{Anopheles gambiae (mosquito)}}\label{bos-taurus}}

We use \emph{Anopheles gambiae} (mosquito), to illustrate the technical steps involved in adding a species to \texttt{stdpopsim}. 
%
%ILAN: commented this out because it is covered by previous sections.
%   the idea is to focus here on the technical demo
%
%In stdpopsim, once all the necessary data (Table 1) for a given species
%is collected, then its inclusion is dependent upon a peer-reviewed
%quality control process. Before this process, it is considered preliminary. 
%If all quality control conditions are met, the
%species and its simulation framework are added to the catalog. Adding a
%species to the stdpopsim catalog is beneficial in many ways: (1) it
%increases the visibility of the species model, (2) it drastically
%improves the reproducibility of the given model framework, and (3) it
%allows other researchers to test different model simulations with that
%particular species.
%
% ILAN: I removed this list to focus on the technical steps
%   assuming that the model has been already figured out from relevant sources.
%   [ this is covered by previous demo on cow ]
%The steps to successfully add a species to the catalog are as follow:
%\begin{itemize}
%\tightlist
%\item
%  Find citeable resources describing the required population and species
%  genetic parameters as detailed in \textbf{Implementing a population genomic simulation}.
%\item
%  Open a GitHub account, fork the stdpopsim GitHub
%  repository, and start a pull request by following the steps provided
%  in the ``Adding a new species'' section of the Development chapter in
%  the stdpopsim docs, currently at
%  %\url{https://popsim-consortium.github.io/stdpopsim-docs/stable/development.html?highlight=adding\%%20species\%20catalog\#adding-a-new-species}
%  These steps as they stand in April 2022 are described in detail in the
%  supplementary material, but are subject to change as the stdpopsim
%  framework improves.
%\end{itemize}
%
\texttt{stdpopsim} uses git for version control. You need to start by opening a GitHub account and forking the \texttt{stdpopsim} repository. Our first step will be to create an upstream
link to the version of the repository owned by
\texttt{popsim-consortium}, then we create a new branch of stdpopsim
using git to keep track of the new species model we are adding.

stdpopsim has a few utilities that interact with Ensembl \citep{ensembl2021},
including retrieving essential genome information to create code
templates for building a new species model. A partial list of the
genomes housed on Ensembl can be found at
https://metazoa.ensembl.org/species.html. (DRS: this is kinda just floating here.)

The next steps flesh out the templates, which include a directory for
the species (AnoGam, for our \emph{Anopheles gamiae} example), and three
files. The first of these files is a data dictionary with space for the assembly
accession number, the assembly name, and the
chromosome names associated with their lengths. The second file contains
species-specific information, including 1. \emph{chromosome-specific}
recombination rates (could also be single rate), 2. a genome-wide
average \emph{mutation rate}, 3. a \emph{generation time} estimate (in
years), 4. a default \emph{effective population size}
used for constant-size simulation, and 5. literature
citations that one can point to for the above and the assembly. The last
file contains the required code for stdpopsim to retrieve the species
information.

The process for adding species data requires editing the template files
with the citeable parameter choices. Here we use the recombination rate
parameter as an example. As a reference for recombination rates in
\emph{Anopheles gambiae} we use a recombination map based on a study
by \citep{Pombi2006}. In that manuscript
the authors cite rates around 1 cM/Mb, with some variation among
arms. This is as easy as editing the following block of code in the
species information file (\texttt{species.py}) that defines the
recombination rate in the simulation:

\begin{verbatim}
_recombination_rate = {
    "2L": 0, # setting to zero because of inversion
    "2R": 1.3e-8,
    "3L": 1.6e-8,
    "3R": 1.3e-8,
    "X": 1e-8,
    "Mt": 0
}
\end{verbatim}

Every piece of information requires a citable reference. To include that we
can create a \texttt{stdpopsim.Citation} object in the same
\texttt{species.py} file. That object looks like this:

\begin{verbatim}
_PombiEtAl = stdpopsim.Citation(
    doi="https://doi.org/10.4269/ajtmh.2006.75.901",
    year=2006,
    author="Pombi et al.",
    reasons={stdpopsim.CiteReason.REC_RATE},
)
\end{verbatim}

stdpopsim in part guarantees code quality through the use of unit
testing, which must be run and passed before starting a pull request to
the stdpopsim github repository.

The full quality control process is fully described at
https://popsim-consortium.github.io/stdpopsim-docs/stable/development.html?highlight=adding\%20species\%20catalog\#demographic-model-review-process.
In summary, we find another stdpopsim contributor (reviewer) to conduct
the quality control process, which consists of: creating a blind
implementation of the model and species to be added; running unit tests
to verify the equivalence of the catalog and quality control model
implementations; and creating a new pull request which is then finally
merged. If conflicts arise between the original and new implementations,
these are discussed and we and the reviewer must come to an agreement.
Outside opinions are useful. Once these conflicts are resolved, the
model graduates from preliminary to a fully vetted model in the stdpopsim catalog.

It is important to note that the catalog is mutable. If more accurate
estimates for a given species are published, then chances are that the
catalog will be updated upon a new quality control review process.

\hypertarget{conclusion}{%
\section*{Conclusion}\label{conclusion}}

As our ability to sequence genomes continues to advance, the need for
population genomic simulation of new model and non-model organism genomes is
becoming acute. So too is the concomitant need for an expandable framework
for implementing such simulations for species of interest, and
the resources for understanding when and how to do so.

In this manuscript we present the basic considerations for implementing
population genomic simulations, agnostic to simulation program. We
describe the steps of determining if a species-specific population
genomic simulation is appropriate for the species and question, what
data is necessary and why, special considerations for finding or using
that data, how to proceed when some of that data is not available,
and why we encourage everyone implementing simulations to have their
parameter choices and implementation reviewed by at least one other
researcher.

We also show how these can be integrated into the stdpopsim catalog, a
resource that is uniquely poised to fill this gap as it provides easy
access to simulations incorporating species-specific information,
easy inclusion of new species genomes, and community-maintained accuracy
and correctness (DRS: not obvious what the difference between accuracy
and correctness is from the context here.).
We additionally briefly describe how the quality control 
process for species inclusion works. Currently, large-scale efforts such as the Earth Biogenome
and its affiliated project networks are generating tens of thousands of genome
assemblies. Each of these assemblies, with some prior knowledge of mutation and
recombination rates, will become a candidate for inclusion into the
stdpopsim catalog following the steps we have outlined above. As
annotation of genome assemblies improves over time those too can easily
be added to the stdpopsim catalog.

Moreover, one of the goals of stdpopsim is to leverage stdpopsim itself
as a springboard for education and inclusion of new communities into
computational biology and software development. We are keen to use
outreach, for instance in the form of workshops and hackathons, as a way
to democratize development of population genetic simulation as well as
grow the stdpopsim catalog and library generally. By enabling
researchers of non-model species with simulation platforms that
traditionally have been quite narrowly focused with respect to organism,
we hope to raise the quality of research across a large number of
systems, while simultanousely expanding the community of software
developers at work in the population and evolutionary genetics world.
Our experience with such outreach over the past two years is that people
are indeed keen to put in the time and effort to include their favority
study species, but that simple, clear guidance is vital. Our
intention with this paper is in part to provide another learning
modality to meet that need. (DRS: these last two sentences very
clearly state one of the major goals of the paper. Maybe they should be
in the intro?)

\hypertarget{acknowledgements}{%
\section*{Acknowledgements}\label{acknowledgements}}

TODO Workshop and hackhaton attendes?

\hypertarget{funding}{%
\section*{Funding}\label{funding}}

TODO: should we order these alphabetically or in the same order as authors?

M. Elise Lauterbur was supported by an NSF Postdoctoral Research Fellowship \#2010884.

Gregor Gorjanc was supported by the University of Edinburgh and BBSRC grant to The Roslin Institute (BBS/E/D/30002275).

Andrew D. Kern and Peter L. Ralph were supported by NIH award R01HG010774.

\bibliography{references}
\end{document}
